launcher/internal/gpu/driverinstaller.go

```diff
@@ -0,0 +1,122 @@
+// Package gpu provides the devices information for cos-gpu-installer
+package gpu
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+	"github.com/google/go-tpm-tools/launcher/spec"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+const (
+	installerContainerID = "tee-gpu-driver-installer-container"
+	installerSnapshotID  = "tee-gpu-driver-installer-snapshot"
+)
+
+// DriverInstaller contains information about the gpu driver installer settings
+type DriverInstaller struct {
+	cdClient   *containerd.Client
+	launchSpec spec.LaunchSpec
+	logger     *log.Logger
+}
+
+// NewDriverInstaller instanciates an object of driver installer
+func NewDriverInstaller(cdClient *containerd.Client, launchSpec spec.LaunchSpec, logger *log.Logger) *DriverInstaller {
+	return &DriverInstaller{
+		cdClient:   cdClient,
+		launchSpec: launchSpec,
+		logger:     logger,
+	}
+}
+
+// InstallGPUDrivers installs the GPU driver on host machine using the cos-gpu-installer container.
+// It uses containerd package to launch and manage the driver installer container.
+func (di *DriverInstaller) InstallGPUDrivers(ctx context.Context) error {
+	ctx = namespaces.WithNamespace(ctx, namespaces.Default)
+	installerImageRef, err := getInstallerImageReference()
+	if err != nil {
+		di.logger.Printf("failed to get the installer container image reference: %v", err)
+		return err
+	}
+
+	di.logger.Printf("cos gpu installer version : %s", installerImageRef)
+	image, err := di.cdClient.Pull(ctx, installerImageRef, containerd.WithPullUnpack)
+	if err != nil {
+		return fmt.Errorf("failed to pull installer image: %v", err)
+	}
+
+	mounts := []specs.Mount{
+		{
+			Type:        "volume",
+			Source:      "/dev",
+			Destination: "/dev",
+			Options:     []string{"rbind", "rw"},
+		}, {
+			Type:        "volume",
+			Source:      "/",
+			Destination: "/root",
+			Options:     []string{"rbind", "rw"},
+		},
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		di.logger.Printf("cannot get hostname: %v", err)
+	}
+
+	container, err := di.cdClient.NewContainer(
+		ctx,
+		installerContainerID,
+		containerd.WithImage(image),
+		containerd.WithNewSnapshot(installerSnapshotID, image),
+		containerd.WithNewSpec(oci.WithImageConfig(image),
+			oci.WithPrivileged,
+			oci.WithProcessArgs("/cos-gpu-installer", "install", "-version=default", fmt.Sprintf("-host-dir=%s", InstallationHostDir)),
```

Comment:
Are you sure it's installing to the right directory? I saw on line 66, it's mounting the root to /root/ in the container

Reply:
root is getting mounted to /root for installer container. we expect drivers to be installed under "/var/lib/nvidia" which is getting set by the host-dir flag where drivers will be installed on the host.